### PR TITLE
Added 6/10 tasks for 27-28

### PR DESCRIPTION
**Pull request by Pozharitskiy**

### CODE REVIEW

File: 27.06/eigth.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function find(array, foo) {
```
Comment (angelinabylitskaya) [inline 27.06/eigth.js:1]: Methods should return value.

File: 27.06/fifth.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function fromPairs(array) {
```
Comment (angelinabylitskaya) [inline 27.06/fifth.js:1]: Try to use "reduce" method here

File: 27.06/nineth.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+function groupBy(array, foo) {
+    const map = new Map();
```
Comment (angelinabylitskaya) [inline 27.06/nineth.js]: Try to use "reduce" method here

File: 27.06/tenth.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+function isEqual(ob1, ob2) {
+   /*Of course :)*/ (JSON.stringify(ob1) === JSON.stringify(ob2)) ? console.log(true) : console.log(false);
```
Comment (angelinabylitskaya) [inline 27.06/tenth.js]: No need to use ?: here.

File: 27.06/third.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+function findIndex(container, place) {
+    const find = (element, index, array) => {
```
Comment (angelinabylitskaya) [inline 27.06/third.js]: Try to use existing js methods

**Review by angelinabylitskaya — CHANGES_REQUESTED** [review]

File: 27.06/tenth.js (hunk 4 above)
Comment (Pozharitskiy) [reply 27.06/tenth.js]: Fixed. Thanks

File: 27.06/eigth.js (hunk 1 above)
Comment (Pozharitskiy) [reply 27.06/eigth.js:1]: Done

File: 27.06/fifth.js (hunk 2 above)
Comment (Pozharitskiy) [reply 27.06/fifth.js:1]: Done

File: 27.06/nineth.js (hunk 3 above)
Comment (Pozharitskiy) [reply 27.06/nineth.js]: Done

File: 27.06/third.js (hunk 5 above)
Comment (Pozharitskiy) [reply 27.06/third.js]: Done

File: 27.06/fifth.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    return array.reduce((current, item) => {
+        let prop = item[0];
```
Comment (kobweb) [inline 27.06/fifth.js]: You should use `const` instead of `let` if your variable wouldn't change.

Comment (kobweb) [reply 27.06/fifth.js]: It related to the whole code.

File: 27.06/fourth.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+function flattenDeep(array) {
+    const lifehack = array.toString().split(",").map(Number);
```
Comment (kobweb) [inline 27.06/fourth.js]: It wouldn't work if you pass to the function something like that [1, ['qwe', [3, ['asd']], 5]]

File: 27.06/nineth.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+function groupBy(array, foo) {
+    var result = array.reduce((elem, currentValue) => {
+        (elem[foo(currentValue)] = elem[foo(currentValue)] || []).push(currentValue);
```
Comment (kobweb) [inline 27.06/nineth.js]: It's unreadable and unsupportable code. Refactor and simplify it.

File: 27.06/tenth.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+function isEqual(ob1, ob2) {
+   /*Of course :)*/return JSON.stringify(ob1) === JSON.stringify(ob2);
```
Comment (kobweb) [inline 27.06/tenth.js]: Won't work if the objects will be the same but with the different fields order.

**Review by kobweb — CHANGES_REQUESTED** [review]

File: 27.06/fifth.js (hunk 6 above)
Comment (Pozharitskiy) [reply 27.06/fifth.js]: Got it

File: 27.06/nineth.js (hunk 8 above)
Comment (Pozharitskiy) [reply 27.06/nineth.js]: Done.

File: 27.06/first.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+function chunk(arr, value) {
+    const array1 = arr.slice(0, value);
+    const array2 = arr.slice(value, arr.length);
+    const container = [];
+    container.push(array1, array2);
+    console.log(container);
```
Comment (ElizavetaSelivanova) [inline 27.06/first.js]: add `return`

File: 27.06/fourth.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+//       console.log(buf);
+//     } 
+
+//     else mas.push(array[i]);
+
+//   }
+
+//   console.log(mas);
+//   console.log(array);
+// 	// TODO: implement function
```
Comment (ElizavetaSelivanova) [inline 27.06/fourth.js]: - add `return`;
- read about `flat` method.

File: 27.06/second.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+function difference(array1, array2) {
+    let uniqueArray = [];
+    const test = array1.filter(element => array2.includes(element));
```
Comment (ElizavetaSelivanova) [inline 27.06/second.js]: incorrect name of variable

File: 27.06/second.js [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+    uniqueArray = uniqueArray.concat(array1, array2);
+    for (let i = 0; i < uniqueArray.length; i++) {
+        uniqueArray[i] === test[0] ? uniqueArray.splice(i, 1) : null;
+    }
+    console.log(uniqueArray);
+}
```
Comment (ElizavetaSelivanova) [inline 27.06/second.js:6]: - add `return`;
- could be optimized.

File: 27.06/seventh.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
+function every(obj, foo) {
+    console.log(obj.every(foo));
+}
```
Comment (ElizavetaSelivanova) [inline 27.06/seventh.js:3]: - add `return`

File: 27.06/sixth.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+            if (array[i] === array[k]) {
+                counter++;
+            }
+        }
+        if (counter === 1) {
+            container.push(array[i]);
+        }
+    }
+    console.log(container);
+}
```
Comment (ElizavetaSelivanova) [inline 27.06/sixth.js:5]: - add `return`;
- could be optimized - no need to use nested `for`

File: 27.06/tenth.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+   console.log(test);
+   var keys = Object.keys(test);
+   var dupe = false;
+   const values = Object.values(test);
+   const isTwo = (el) => {
+      return (el === 2);
+   }
+   (values.every(isTwo) ? console.log(true) : console.log(false))
+}
```
Comment (ElizavetaSelivanova) [inline 27.06/tenth.js:32]: - the solution should be simplified and more readable;
- read about `let`, `const` and difference from `var`;
- review naming of variables.

File: 27.06/tenth.js (hunk 9 above)
Comment (Pozharitskiy) [reply 27.06/tenth.js]: Solved

File: 27.06/second.js (hunk 12 above)
Comment (Pozharitskiy) [reply 27.06/second.js]: Fixed

File: 27.06/second.js (hunk 13 above)
Comment (Pozharitskiy) [reply 27.06/second.js:6]: oh, it can  be simpler. Fixed

File: 27.06/first.js (hunk 10 above)
Comment (Pozharitskiy) [reply 27.06/first.js]: Done

File: 27.06/fourth.js (hunk 7 above)
Comment (Pozharitskiy) [reply 27.06/fourth.js]: Fixed

File: 27.06/fourth.js (hunk 11 above)
Comment (Pozharitskiy) [reply 27.06/fourth.js]: Thanks! It was helpful. Done

File: 27.06/seventh.js (hunk 14 above)
Comment (Pozharitskiy) [reply 27.06/seventh.js:3]: Done

File: 27.06/sixth.js (hunk 15 above)
Comment (Pozharitskiy) [reply 27.06/sixth.js:5]: I found more elegant solution. Check it out please

File: 27.06/tenth.js (hunk 16 above)
Comment (Pozharitskiy) [reply 27.06/tenth.js:32]: 1) I added comments to the solution. Hope it will be more readable.
2) I know about that, but ... (my bad) Fixed
3) Done

**Pull request merged**